mpd/client.py

```diff
@@ -217,168 +217,93 @@ def _init():
 
     return client, conn, cur, current_song_coords
 
-
-def main_album(queue_length):
+def main_album(queue_length, option_best):
     client, conn, cur, current_song_coords = _init()
 
+    # Get 'queue_length' random albums
     for i in range(queue_length):
         # No cache management
         # Get all songs from the current album
-        album = current_song_coords["album"]
-        cur.execute("SELECT id, tempo, amplitude, frequency, attack, filename, album FROM songs WHERE album=?", (album,))
+        distance_array = []
+
+        # Get album name and all of this album's songs coordinates
+        album_name = current_song_coords["album"]
+        cur.execute("SELECT id, tempo, amplitude, frequency, attack, filename, album FROM songs WHERE album=?", (album_name,))
         target_album_set = cur.fetchall()
 
-        # Get all other songs
-        cur.execute("SELECT id, tempo, amplitude, frequency, attack, filename, album FROM songs ORDER BY album")
-        tmp_song_data = cur.fetchone()
-        shortest_distance = -1
-
-        # Check the best suitable album
-        while tmp_song_data:
-            current_album_set = list()
-            current_album_set.append(tmp_song_data)
-            tmp_song_data = cur.fetchone()
-
-            i = 0
-            # Get all songs from the current temporary album
-            while tmp_song_data:
-                if (current_album_set[i]["album"] == tmp_song_data["album"]):
-                    current_album_set.append(tmp_song_data)
-                else:
-                    break
-                tmp_song_data = cur.fetchone()
-                i = i + 1
-            # Skip current album and already processed albums
-            if((current_album_set[0]["album"] != target_album_set[0]["album"]) and
-               not (("file: %s" % (current_album_set[0]["filename"],)) in client.playlist())):
-                tmp_distance = distance_sets(current_album_set, target_album_set)
-                if tmp_distance < shortest_distance or shortest_distance == -1:
-                    shortest_distance = tmp_distance
-                    closest_album = current_album_set
-
-        logging.info("Closest album found is \"%s\". Distance is %f." % (closest_album[0]["album"], shortest_distance))
-        for song in closest_album:
+        # Get all albums
+        cur.execute("SELECT DISTINCT album FROM songs")
+        albums = cur.fetchall();
+
+        # Compute the distance between current album and all other albums
+        for tmp_album in albums:
```

Comment:
Better to use `for tmp_album in cur.execute()` which uses an iterator. See 5th example on https://docs.python.org/2/library/sqlite3.html.